geode-gfsh/src/test/java/org/apache/geode/management/internal/cli/result/TableBuilderTest.java

```diff
@@ -0,0 +1,335 @@
+
+    List<String> result = validateTable(table, true);
+
+    assertThat(result)
+        .hasSize(4)
+        .containsExactly(
+            "",
+            "Field0<|>  Field1  <|>  Field2  <|>Fie..",
+            "------<|>----------<|>----------<|>-----",
+            "1     <|>123456789-<|>123456789-<|>123..");
+  }
+
+  @Test
+  public void testManyColumns() {
+    Table table = createTableStructure(8, "|", fill(new String[8], "Field"));
+    RowGroup rowGroup = table.getLastRowGroup();
+    Row row = rowGroup.newRow();
+    row
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-");
+
+    List<String> result = validateTable(table, true);
+
+    assertThat(result)
+        .hasSize(4)
+        .containsExactly(
+            "",
+            "  Field0  |  Field1  |..|..|..|..|..|..",
+            "----------|----------|--|--|--|--|--|--",
+            "123456789-|123456789-|..|..|..|..|..|..");
+  }
+
+  @Test
+  public void testDisableColumnAdjustment() {
+    when(screen.shouldTrimColumns()).thenReturn(false);
+
+    Table table = createTableStructure(5, "|", fill(new String[5], "Field"));
+    RowGroup rowGroup = table.getLastRowGroup();
+    Row row = rowGroup.newRow();
+    row
+        .newLeftCol("1")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-")
+        .newLeftCol("123456789-123456789-12345")
+        .newLeftCol("1");
+
+    List<String> result = validateTable(table, false);
+
+    assertThat(result)
+        .hasSize(4)
+        .containsExactly(
+            "",
+            "Field0|  Field1  |  Field2  |         Field3          |Field4",
+            "------|----------|----------|-------------------------|------",
+            "1     |123456789-|123456789-|123456789-123456789-12345|1");
+  }
+
+  private Table createTableStructure(int columnCount, String separator, String... columnNames) {
+    Table resultTable = tableBuilder.newTable(screen);
+    resultTable.setTabularResult(true);
+    resultTable.setColumnSeparator(separator);
+
+    resultTable.newBlankRow();
+    RowGroup rowGroup = resultTable.newRowGroup();
+    Row row = rowGroup.newRow();
+
+    for (int column = 0; column < columnCount; column++) {
+      row.newCenterCol(columnNames[column] + column);
+    }
+
+    rowGroup.newRowSeparator('-', false);
+
+    return resultTable;
+  }
+
+  private List<String> validateTable(Table table, boolean shouldTrim) {
+    String tableAsString = table.buildTable();
+    System.out.println(tableAsString);
+
+    List<String> lines = asList(tableAsString.split(GfshParser.LINE_SEPARATOR));
+
+    int lineCount = 0;
+    for (String line : lines) {
+      System.out.println("Line #" + lineCount++ + ": length = " + line.length() + ", isWider = "
```

Comment:
See above comment regarding print statements. Perhaps a suitably constructed `as()` call added to the below assertions would render this unnecessary.